frontend/public/components/autocomplete.tsx

```diff
@@ -0,0 +1,134 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import * as classNames from 'classnames';
+import { TextInput } from '@patternfly/react-core';
+import { useDocumentListener, getLabelsAsString } from '@console/shared';
+import { KeyEventModes } from '@console/shared/src/hooks';
+import { fuzzyCaseInsensitive } from './factory/table-filters';
+import { K8sResourceKind } from '../module/k8s';
+
+const MAX_SUGGESTIONS = 5;
+
+const labelParser = (resource: K8sResourceKind[]) => {
```

Comment:
Name this `resources` to make it obvious there's more than one. Probably better to make this `K8sResourceCommon` as well.

```suggestion
const labelParser = (resources: K8sResourceCommon[]): Set<string> => {
```